scipy/optimize/_bracket.py

```diff
@@ -435,13 +435,13 @@ def _bracket_minimum_iv(func, xm0, xl0, xr0, xmin, xmax, factor, args, maxiter):
     if xl0_not_supplied:
         xl0 = xl0.copy()
         cond = ~np.isinf(xmin) & (xl0 < xmin)
-        xl0[cond] = (
+        xl0[cond] = xm0[cond] - (
             xm0[cond] - xmin[cond]
         ) / np.array(16, dtype=xl0.dtype)
```

Comment:
Just to be sure I'm reading this correctly this time, here is my understanding of the logic if `xl0` was not provided:

- Above, we assigned a value to `xl0` blindly, just so we can get everything to be arrays of the same shape.
- `cond` is only true where `xl0 < xmin`, which means that `xmin` is very close to (`<0.5` from) `xm0`.
- Where `cond` is true, we replace our blind value of `xl0` with a value `d/16` from `x0`, where `d` is the distance between `xm0` and `min`. This means that `d < 0.03125`.

If that's right:
- Re: `cond = ~np.isinf(xmin) & (xl0 < xmin)`. Is it possible for the second condition to by true without the first? If not, couldn't this just be `cond = xl0 < xmin`?
- Right around `xmin = xm0 - 0.5`, `xl0` will jump from being `xm0 - 0.5` to `xm0 - 0.3125`. I'm sure this is fine in practice; just a little weird.
- Not that it needs to move, but *could* this logic be moved up before all the broadcasting?

For instance, above, could it be:
```
    if xl0 is None:
        xl0 = xm0 - np.minimum( (xm0 - xmin) / 16, 0.5)
```
(if this is similar logic? except that it would eliminate the jump)

The downside is that you could get an error since you haven't checked whether `xmin` is numeric yet, but I think a little shuffling could fix that.

Even if this is correct, no need to change it - since it probably doesn't matter in practice. Part of the purpose of thinking it through is to make sure I understand correctly this time.

Reply:
Thanks @mdhaber. I think you understand everything well.

> * cond = ~np.isinf(xmin) & (xl0 < xmin)

I'm trying to think if there's a reason why I wrote this in such a way. My guess is just that I wasn't paying attention.

> * Right around `xmin = xm0 - 0.5`, `xl0` will jump from being `xm0 - 0.5` to `xm0 - 0.3125`. I'm sure this is fine in practice; just a little weird.

I also just wasn't paying attention here. Your suggestion using `np.minimum` looks better to me to. I'll change to that.

> * Not that it needs to move, but _could_ this logic be moved up before all the broadcasting?

I can't remember the details, but I tried to make this work with this logic before the broadcasting and gave up because I didn't want to sink too much time into it. I think the only issues were with input validation like you mentioned. It should be possible to make it work and I can do that in a followup.

Reply:
Don't worry about moving it. Maybe just put dummy values - not even `xm0 - 0.5` - into the array to begin with so it's obvious that it's just for the sake of broadcasting, then use `np.minimum` after to compute the real values. Hopefully that will free a little time to work on replacing that input validation error message with NaNs and appropriate status indicator in the result object. Thanks!

Reply:
> Don't worry about moving it. Maybe just put dummy values - not even `xm0 - 0.5` - into the array to begin with so it's obvious that it's just for the sake of broadcasting, then use `np.minimum` after to compute the real values. Hopefully that will free a little time to work on replacing that input validation error message with NaNs and appropriate status indicator in the result object. Thanks!

Done! Things are much cleaner after your suggestions. I'll follow up with the input validation thing later this week.